discovery/manager.go

```diff
@@ -153,41 +146,41 @@ func (m *Manager) cancelDiscoverers() {
 	for _, c := range m.discoverCancel {
 		c()
 	}
-	m.targets = make(map[poolKey][]*targetgroup.Group)
+	m.targets = make(map[poolKey]map[string]*targetgroup.Group)
 	m.discoverCancel = nil
 }
 
-func (m *Manager) addGroup(poolKey poolKey, tg []*targetgroup.Group) {
+func (m *Manager) updateGroup(poolKey poolKey, tgs []*targetgroup.Group) {
 	done := make(chan struct{})
 
 	m.actionCh <- func(ctx context.Context) {
-		if tg != nil {
-			m.targets[poolKey] = tg
+		if tgs == nil {
```

Comment:
We don't need this check right?

Reply:
I think we do. 
it fixes this bug https://github.com/prometheus/prometheus/issues/3653
k8s was sending `nil` and it panicked.

this is fixed in https://github.com/prometheus/prometheus/pull/3660  , but I prefer to keep one more check just in case any other Discoverer decided to do the same.

Reply:
@Gouthamve , oops , that was for the line below: `if tg != nil {` 
I have removed this check.
I wasn't sure what happens when you range over nil maps, but just now did a quick check and it doesn't cause any panics so safe to remove